flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/hybrid/tiered/tier/remote/RemoteTierFactory.java

```diff
@@ -90,7 +91,12 @@ public TierProducerAgent createProducerAgent(
     public TierConsumerAgent createConsumerAgent(
             List<TieredStorageConsumerSpec> tieredStorageConsumerSpecs,
             TieredStorageNettyService nettyService) {
-        // TODO, implement the remote tier consumer agent
-        return null;
+        PartitionFileReader partitionFileReader =
+                SegmentPartitionFile.createPartitionFileReader(
+                        getTieredStoragePath(remoteStorageBasePath));
+        return new RemoteTierConsumerAgent(
+                new RemoteStorageScanner(getTieredStoragePath(remoteStorageBasePath)),
+                partitionFileReader,
+                bufferSizeBytes);
```

Comment:
Minor:
1. `getTieredStoragePath(remoteStorageBasePath)` can be deduplicated
2. It might be better to align code-style for `partitionFileReader` and `remoteStorageScanner`, i.e. creating them either both inline when creating the agent, or both before creating the agent.